server/src/util.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  Diagnostic,
+  DiagnosticSeverity,
+} from 'vscode-languageserver';
+import { IRuleResult } from '@stoplight/spectral';
+import { DiagnosticSeverity as SpectralDiagnosticSeverity } from '@stoplight/types/dist/diagnostics';
+
+/**
+ * Converts a Spectral rule violation severity into a VS Code diagnostic severity.
+ * @param {SpectralDiagnosticSeverity} severity - The Spectral diagnostic severity to convert.
+ * @return {DiagnosticSeverity} The converted severity for a VS Code diagnostic.
+ */
+function convertSeverity(severity: SpectralDiagnosticSeverity): DiagnosticSeverity {
+  switch (severity) {
+    case SpectralDiagnosticSeverity.Error:
+      return DiagnosticSeverity.Error;
+    case SpectralDiagnosticSeverity.Warning:
+      return DiagnosticSeverity.Warning;
+    case SpectralDiagnosticSeverity.Information:
+      return DiagnosticSeverity.Information;
+    case SpectralDiagnosticSeverity.Hint:
+      return DiagnosticSeverity.Hint;
+    default:
+      return DiagnosticSeverity.Error;
+  }
+}
+
+/**
+ * Converts a Spectral rule violation to a VS Code diagnostic.
+ * @param {IRuleResult} problem - The Spectral rule result to convert to a VS Code diagnostic message.
+ * @return {IDiagnostic} The converted VS Code diagnostic to send to the client.
+ */
+export function makeDiagnostic(problem: IRuleResult): Diagnostic {
+  return {
+    range: {
+      start: {
+        line: problem.range.start.line,
```

Comment:
Does LSP expect 1-based or 0-based lines & columns?
Don't recall to be honest. I know VS Code's Monaco demands 1-based values.

Reply:
@P0lip Those work as-is. VSCode seems to properly highlight the proper ranges. Thnx for having raised this!